docs/graphs/classdiagram_lib.dot

```diff
@@ -0,0 +1,34 @@
+digraph hierarchy {
+size="5,5"
+node[   shape=record,
+        style=filled,
+        fillcolor=gray95]
+edge[dir=back, arrowtail=none]
+
+
+ps[label = "{ProgramState|
+                all_transactions\l
+                filtered_transactions\l|
+                + setup()\l
+                + add(dataframe)\l
+                + apply(filter)\l
+                + categorize(row, category)\l
+                + store()\l
+                }"]
+imp[label = "{TextImporter|| + read(file) : dataframe}"]
+tfd[label = "{TextfileDescription|
```

Comment:
What is the purpose of `TextfileDescription`?

Reply:
This would contain the specifics for one type of input file. Eg. how do the columns map to our internal columns, what is the CsvImporter dialect

Reply:
For clarity I think I can just commit an example implementation